scalafix/rules/src/main/scala/scalafix/internal/rule/RscCompat.scala

```diff
@@ -0,0 +1,537 @@
+// seems to be possible.
+
+class TypePrinter(env: Env) {
+  private val baos = new ByteArrayOutputStream
+  private val out = new PrintStream(baos, true, UTF_8.name)
+  override def toString = new String(baos.toByteArray, UTF_8)
+
+  def pprint(tpe: s.Type): Unit = {
+    def prefix(tpe: s.Type): Unit = {
+      tpe match {
+        case s.TypeRef(pre, sym, args) =>
+          if (sym.startsWith("scala/Function") &&
+              args.exists(_.isInstanceOf[s.ByNameType])) {
+            var params :+ ret = args
+            if (params.length != 1) out.print("(")
+            rep(params, ", ") { param =>
+              // FIXME: https://github.com/twitter/rsc/issues/142
+              out.print("(")
+              normal(param)
+              out.print(")")
+            }
+            if (params.length != 1) out.print(")")
+            out.print(" => ")
+            normal(ret)
+          } else {
+            val prettyPre = if (pre == s.NoType) sym.trivialPrefix else pre
+            prettyPre match {
+              case _: s.SingleType | _: s.ThisType | _: s.SuperType =>
+                prefix(prettyPre)
+                out.print(".")
+              case s.NoType =>
+                ()
+              case _ =>
+                prefix(prettyPre)
+                out.print("#")
+            }
+            pprint(sym)
+            rep("[", args, ", ", "]")(normal)
+          }
+        case s.SingleType(pre, sym) =>
+          val prettyPre = if (pre == s.NoType) sym.trivialPrefix else pre
+          opt(prettyPre, ".")(prefix)
+          pprint(sym)
+        case s.ThisType(sym) =>
+          opt(sym, ".")(pprint)
+          out.print("this")
+        case s.SuperType(pre, sym) =>
+          opt(pre, ".")(prefix)
+          out.print("super")
+          opt("[", sym, "]")(pprint)
+        case s.WithType(types) =>
+          rep(types, " with ") { tpe =>
+            // FIXME: https://github.com/twitter/rsc/issues/142
+            val needsParens = tpe.isInstanceOf[s.ExistentialType]
+            if (needsParens) out.print("(")
+            normal(tpe)
+            if (needsParens) out.print(")")
+          }
+        case s.StructuralType(utpe, decls) =>
+          decls.infos.foreach(notes.append)
+          opt(utpe)(normal)
+          if (decls.infos.nonEmpty) {
+            rep(" { ", decls.infos, "; ", " }")(pprint)
+          } else {
+            utpe match {
+              case s.WithType(tpes) if tpes.length > 1 => ()
+              case _ => out.print(" {}")
+            }
+          }
+        case s.AnnotatedType(anns, utpe) =>
+          opt(utpe)(normal)
+          out.print(" ")
+          rep(anns, " ", "")(pprint)
+        case s.ExistentialType(utpe, decls) =>
+          decls.infos.foreach(notes.append)
+          opt(utpe)(normal)
+          rep(" forSome { ", decls.infos, "; ", " }")(pprint)
+        case s.UniversalType(tparams, utpe) =>
+          out.append("({ type λ")
```

Comment:
My understanding is "λ" is just a local type variable that the type lambda is bound to, but other names could be used instead. So that means the generated code could be stylistically inconsistent with other uses of type lambdas in the codebase, right? It would be worth noting that in a `FIXME` comment here.

(My guess is it would make more sense to make naming configurable at some point rather than trying to automatically infer the naming convention in a given codebase.)

Reply:
`λ` is a somewhat idiomatic to call this helper type alias, though it's definitely not a standard. As you said, it would be good to have this configurable.